core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -297,14 +336,60 @@ private void createSchemaTopic() throws StoreInitializationException {
                + desiredReplicationFactor + ". If this is a production environment, it's "
                + "crucial to add more brokers and increase the replication factor of the topic.");
     }
-    Properties schemaTopicProps = new Properties();
-    schemaTopicProps.put(LogConfig.CleanupPolicyProp(), "compact");
 
+    NewTopic schemaTopicRequest = new NewTopic(topic, 1, (short) schemaTopicReplicationFactor);
+    schemaTopicRequest.configs(Collections.singletonMap(LogConfig.CleanupPolicyProp(), "compact"));
     try {
-      AdminUtils.createTopic(zkUtils, topic, 1, schemaTopicReplicationFactor, schemaTopicProps,
-                             RackAwareMode.Enforced$.MODULE$);
-    } catch (TopicExistsException e) {
-      // This is ok.
+      admin.createTopics(Collections.singleton(schemaTopicRequest)).all().get(30, TimeUnit.SECONDS);
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof TopicExistsException) {
+        // This is ok.
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  private void verifySchemaTopic(AdminClient admin) throws StoreInitializationException,
+                                                           InterruptedException,
+                                                           ExecutionException,
+                                                           TimeoutException {
+    log.info("Validating schemas topic {}", topic);
+
+    Set<String> topics = Collections.singleton(topic);
+    Map<String, TopicDescription> topicDescription = admin.describeTopics(topics)
+        .all().get(30, TimeUnit.SECONDS);
+
+    TopicDescription description = topicDescription.get(topic);
+    if (description.partitions().size() != 1) {
+      log.error("The schema topic " + topic + " should have only 1 partition.");
+    }
+
+    if (description.partitions().get(0).replicas().size() < desiredReplicationFactor) {
+      log.warn("The replication factor of the schema topic "
+               + topic
+               + " is less than the desired one of "
+               + desiredReplicationFactor
+               + ". If this is a production environment, it's crucial to add more brokers and "
+               + "increase the replication factor of the topic.");
+    }
+
+    ConfigResource topicResource = new ConfigResource(ConfigResource.Type.TOPIC, topic);
+
+    Map<ConfigResource, Config> configs =
+        admin.describeConfigs(Collections.singleton(topicResource)).all().get(30, TimeUnit.SECONDS);
+    Config topicConfigs = configs.get(topicResource);
+    String retentionPolicy = topicConfigs.get(LogConfig.CleanupPolicyProp()).value();
+    if (retentionPolicy == null || "compact".compareTo(retentionPolicy) != 0) {
```

Comment:
`!retentionPolicy.equals("compact")` instead of `"compact".compareTo(retentionPolicy) != 0`?